crates/chain/src/chain_data.rs

```diff
@@ -192,4 +205,88 @@ mod test {
             "confirmation_height is higher then it should be higher ord"
         );
     }
+
+    #[test]
+    fn test_sort_unconfirmed_chain_position() {
+        let mut v = vec![
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: Some(5),
+                last_seen: Some(20),
+            },
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: None,
+                last_seen: Some(30),
```

Comment:
It's impossible to have a first-seen and no last-seen. The same goes vise-versa.